crates/kornia-pnp/src/ippe.rs

```diff
@@ -0,0 +1,170 @@
+//! Infinitesimal Plane-based Pose Estimation (IPPE)
+//!
+//! This module provides routines to estimate two candidate poses for a planar
+//! object from 2D to 3D correspondences, following the IPPE method.
+//!
+//! References:
+//! - T. Collins and A. Bartoli, "Infinitesimal Plane-based Pose Estimation"
+//! - OpenCV implementation reference: modules/calib3d/src/ippe.hpp/.cpp
+//!
+//! This file contains original Rust code inspired by the algorithms described
+//! in the references above. The OpenCV implementation is BSD-3-Clause; see the
+//! OpenCV project license for details.
+
+use crate::pnp::{PnPError, PnPResult};
+use glam::{Mat3A, Vec3, Vec3A};
+use kornia_lie::so3::SO3;
+
+/// Two IPPE pose solutions sorted by reprojection error (best first).
+///
+/// Note: The current implementation returns the same solution twice `(best, second)`
+/// as a temporary placeholder until the IPPE-specific second solution is implemented.
+pub type IPPEResult = (PnPResult, PnPResult);
+
+/// Marker type for the IPPE solver.
+pub struct IPPE;    
+
+impl IPPE {
```

Comment:
if this is pretended to be a stateless struct I think this should be a freee function `solve_pnp_ippe` without the struct